netty/src/main/java/io/grpc/transport/netty/NettyClientHandler.java

```diff
@@ -280,32 +319,61 @@ protected boolean isGracefulShutdownComplete() {
    * the creation request is queued.
    */
   private void createStream(CreateStreamCommand command, final ChannelPromise promise) {
-    final int streamId = getAndIncrementNextStreamId();
+    // Get the stream ID for the new stream.
+    final int streamId;
+    try {
+      streamId = getAndIncrementNextStreamId();
+    } catch (StatusException e) {
+      // Stream IDs have been exhausted for this connection. Fail the promise immediately.
+      promise.setFailure(e);
+      return;
+    }
+
     final NettyClientStream stream = command.stream();
     final Http2Headers headers = command.headers();
-    // TODO: Send GO_AWAY if streamId overflows
```

Comment:
Why remove this TODO? We still aren't stopping the transport if streamId overflowed.